File: src/main/java/uk/co/threebugs/mochiwhattotrade3/account/AccountRepository.java
package uk.co.threebugs.mochiwhattotrade3.account;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.query.QueryByExampleExecutor;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface AccountRepository extends JpaRepository<Account, Integer>, QueryByExampleExecutor<Account> {
    Optional<Account> findByName(String name);
}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/account/AccountRepository.kt
package uk.co.threebugs.mochiwhattotrade3.account

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.query.QueryByExampleExecutor
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface AccountRepository : JpaRepository<Account, Int>, QueryByExampleExecutor<Account> {
    fun findByName(name: String): Optional<Account>
}
